core/src/main/java/org/bitcoinj/wallet/DerivationPathFactory.java

```diff
@@ -56,35 +56,66 @@ public ImmutableList<ChildNumber> identityInvitationFundingDerivationPath() {
                 .build();
     }
 
+    /** blockchain identity keys derivation root path (EC Keys)
+     * m/9'/5'/5'/0'/0'/(blockchain identity index)' (mainnet)
+     * m/9'/1'/5'/0'/0'/(blockchain identity index)' (testnet, devnets)
+     */
+    @Deprecated
+    public ImmutableList<ChildNumber> blockchainIdentityECDSADerivationPath() {
+        return ImmutableList.<ChildNumber>builder()
+                .add(FEATURE_PURPOSE)
+                .add(coinType)
+                .add(FEATURE_PURPOSE_IDENTITIES)
+                .add(ChildNumber.ZERO_HARDENED) //sub feature 0
+                .add(ChildNumber.ZERO_HARDENED) //key type (0 is ECDSA key)
+                .add(ChildNumber.ZERO_HARDENED) //identity index (default to 0 for now)
+                .build();
+    }
+
     /** blockchain identity keys derivation path (EC Keys)
      * m/9'/5'/5'/0'/0'/(blockchain identity index)'/(key index)' (mainnet)
      * m/9'/1'/5'/0'/0'/(blockchain identity index)'/(key index)' (testnet, devnets)
      */
-    public ImmutableList<ChildNumber> blockchainIdentityECDSADerivationPath() {
+    public ImmutableList<ChildNumber> blockchainIdentityECDSADerivationPath(int index) {
         return ImmutableList.<ChildNumber>builder()
                 .add(FEATURE_PURPOSE)
                 .add(coinType)
                 .add(FEATURE_PURPOSE_IDENTITIES)
                 .add(ChildNumber.ZERO_HARDENED) //sub feature 0
                 .add(ChildNumber.ZERO_HARDENED) //key type (0 is ECDSA key)
                 .add(ChildNumber.ZERO_HARDENED) //identity index (default to 0 for now)
-                .add(ChildNumber.ZERO_HARDENED) //key index (default to 0 for now)
+                .add(new ChildNumber(index, true)) //key index (default to 0 for now)
+                .build();
+    }
+
+    /** blockchain identity keys derivation root path (BLS Keys)
+     * m/9'/5'/5'/0'/1'/(blockchain identity index = 0) (mainnet)
+     * m/9'/1'/5'/0'/1'/(blockchain identity index = 0) (testnet, devnets)
+     */
+    public ImmutableList<ChildNumber> blockchainIdentityBLSDerivationPath() {
+        return ImmutableList.<ChildNumber>builder()
+                .add(FEATURE_PURPOSE)
+                .add(coinType)
+                .add(FEATURE_PURPOSE_IDENTITIES)
+                .add(ChildNumber.ZERO_HARDENED) //sub feature 0
+                .add(ChildNumber.ONE_HARDENED) //key type (0 is ECDSA key)
```

Comment:
Should this comment here still say `0 is ECDSA key`?

Reply:
The comment should say `(1 is BLS key)`